bento/crates/workflow/src/tasks/resolve.rs

```diff
@@ -32,16 +36,52 @@ pub async fn resolver(agent: &Agent, job_id: &Uuid, request: &ResolveReq) -> Res
             conditional_receipt.claim.clone().as_value()?.output.as_value()?
         {
             if !guest_output.assumptions.is_empty() {
-                let assumptions = guest_output.assumptions.as_value()?.iter();
+                let assumptions = guest_output
+                    .assumptions
+                    .as_value()
+                    .context("Failed unwrap the assumptions of the guest output")?
+                    .iter();
+
                 tracing::info!("Resolving {} assumption(s)", assumptions.len());
-                assumptions_len = Some(assumptions.len().try_into()?);
+                assumptions_len =
+                    Some(assumptions.len().try_into().context("Failed to convert to i64")?);
```

Comment:
```suggestion
                    Some(assumptions.len().try_into().context("Failed to convert to u64")?);
```

(or some other error message)